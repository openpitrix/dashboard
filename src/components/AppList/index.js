import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import Card from '../Card';
import styles from './index.scss';

export default class AppList extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    fold: PropTypes.bool,
  }

  render() {
    const { apps, fold, className } = this.props;

    return (
      <div className={classnames(styles.appList, className)}>
        {
          apps && apps.map(app =>
            <Link key={app.id} to={`/app/${app.id}`}>
              <Card
                icon={app.icon}
                name={app.name}
                desc={app.description}
                fold={fold}
              />
            </Link>,
          )
        }
      </div>
    );
  }
}
