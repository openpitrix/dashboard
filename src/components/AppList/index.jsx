import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { translate } from 'react-i18next';

import Card from '../Card';
import CardTitle from '../Card/CardTitle';

import styles from './index.scss';

@translate()
export default class AppList extends PureComponent {
  static propTypes = {
    appSearch: PropTypes.string,
    apps: PropTypes.array,
    categoryApps: PropTypes.array,
    categoryTitle: PropTypes.string,
    className: PropTypes.string,
    isLoading: PropTypes.bool,
    skipLink: PropTypes.string
  };

  static defaultProps = {
    apps: [],
    categoryApps: [],
    skipLink: 'apps',
    isLoading: false
  };

  renderCategoryApps() {
    const {
      categoryTitle, appSearch, categoryApps, skipLink
    } = this.props;
    const categoryShow = !categoryTitle && !appSearch && categoryApps;

    if (!categoryShow) {
      return null;
    }

    if (categoryShow) {
      return categoryApps.map(({ category_id, name, apps }) => {
        apps = apps || [];
        if (apps.length === 0) {
          return null;
        }

        return (
          <div key={category_id} className={styles.categoryApps}>
            {apps.length > 0 && (
              <CardTitle
                skipLink={skipLink}
                categoryId={category_id}
                title={name}
                more
              />
            )}

            {apps
              .slice(0, 6)
              .map(app => (
                <Card
                  key={app.app_id}
                  icon={app.icon}
                  name={app.name}
                  desc={app.description}
                  link={`/${skipLink}/${app.app_id}`}
                  fold
                />
              ))}
          </div>
        );
      });
    }
  }

  getSearchTitle() {
    const { appSearch, categoryTitle, t } = this.props;
    return (appSearch && t('search_results')) || categoryTitle || t('Newest');
  }

  render() {
    const {
      apps, className, skipLink, isLoading, t
    } = this.props;

    return (
      <div className={classnames(className)}>
        <div className={styles.appList}>
          {<CardTitle title={this.getSearchTitle()} more={false} />}

          {apps.map((app, idx) => (
            <Card
              icon={app.icon}
              name={app.name}
              desc={app.description}
              key={idx}
              link={`/${skipLink}/${app.app_id}`}
            />
          ))}

          {!apps.length
            && !isLoading && (
              <div className={styles.noData}>{t('NO_SEARCH_DATA')}</div>
          )}
        </div>

        {this.renderCategoryApps()}
      </div>
    );
  }
}
