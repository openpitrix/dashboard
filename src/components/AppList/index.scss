@import '~scss/vars';

.appList {
  a {
    &:nth-child(3n+4) {
      &>div {
        margin-right: 0;
      }
    }
  }
  .noData{
    padding: 20px 0;
    text-align: center;
    font-size: 12px;
    color: $N75;
  }
}
.categoryApps{
  margin-top: -16px;
}

.appsRow {
  display: flex;
  justify-content: space-between;
}
