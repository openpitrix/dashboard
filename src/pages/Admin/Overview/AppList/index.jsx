import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { translate } from 'react-i18next';

import TdName from 'components/TdName/index';
import { imgPlaceholder } from 'utils';

import styles from './index.scss';

@translate()
export default class AppList extends PureComponent {
  static propTypes = {
    apps: PropTypes.array,
    isAdmin: PropTypes.bool
  };

  render() {
    const { apps, isAdmin, t } = this.props;
    const imgPhd = imgPlaceholder(24);

    return (
      <ul className={classNames(styles.appList, { [styles.normalList]: !isAdmin })}>
        {apps.map((data, index) => (
          <li key={data.app_id}>
            {isAdmin && <span className={styles.order}>{index + 1}</span>}
            <TdName
              image={data.icon || imgPhd}
              name={data.name}
              description={data.description}
              linkUrl={`/dashboard/app/${data.app_id}`}
              noCopy={true}
            />
          </li>
        ))}
      </ul>
    );
  }
}
