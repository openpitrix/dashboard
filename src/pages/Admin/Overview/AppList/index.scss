@import '~scss/vars';

.appList{
  li{
    box-sizing: border-box;
    padding: 12px 18px 13px 16px;
    height: 60px;
    border-bottom: 1px solid $border-gray-li;
    &:hover{
      background-color: $li-hover-bg;
      [class*="tdName"] a{
        color: $link-hover;
      }
    }
  }
  .order{
    display: inline-block;
    float: left;
    margin-right: 18px;
    font-size: $font16;
    font-weight: 500;
    line-height: 1.25;
    color: $N500;
  }
  .total{
    display: inline-block;
    float: right;
    margin-top: 10px;
    font-size: 12px;
    text-align: left;
    color: $grey-word;
    .number{
      display: inline-block;
      margin-right: 4px;
      width: 30px;
      height: 20px;
      line-height: 20px;
      border-radius: 2px;
      background-color: $background-color;
      font-size: 12px;
      font-weight: 500;
      text-align: center;
      color: $N500;
    }
  }
}
.normalList{
  li{
    margin-bottom: 8px;
  }
}
