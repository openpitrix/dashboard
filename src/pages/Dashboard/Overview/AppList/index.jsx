import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import TdName from 'components/TdName/index';
import { getFilterObj } from 'utils';
import styles from './index.scss';

@translate()
export default class AppList extends PureComponent {
  static propTypes = {
    apps: PropTypes.array,
    isAdmin: PropTypes.bool,
    isDev: PropTypes.bool,
    topApps: PropTypes.array
  };

  static defaultProps = {
    topApps: [],
    apps: [],
    isAdmin: false,
    isDev: false
  };

  render() {
    const {
      topApps, apps, isAdmin, isDev
    } = this.props;
    const items = isAdmin ? topApps : apps;
    return (
      <ul className={styles.appList}>
        {items.map((item, index) => {
          const app = getFilterObj(apps, 'app_id', item.id);
          if (isAdmin) {
            item.app_id = item.id;
          }

          return (
            <li key={item.id || item.app_id}>
              {isAdmin && <span className={styles.order}>{index + 1}</span>}
              <TdName
                className="largeShow"
                image={app.icon || item.icon}
                imageSize={24}
                name={item.name || app.name}
                description={item.description || app.description}
                linkUrl={
                  isDev
                    ? `/dashboard/app/${item.app_id}`
                    : `/store/${item.app_id}`
                }
                noCopy={true}
              />
            </li>
          );
        })}
      </ul>
    );
  }
}
