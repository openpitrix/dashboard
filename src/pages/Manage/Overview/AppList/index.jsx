import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import TdName from 'components/TdName/index';
import styles from './index.scss';

export default class AppList extends PureComponent {
  static propTypes = {
    apps: PropTypes.array
  };

  render() {
    const { apps } = this.props;
    return (
      <ul className={styles.appList}>
        {apps.map((data, index) => (
          <li key={data.app_id}>
            <span className={styles.order}>{index + 1}</span>
            <TdName image={data.icon} name={data.name} description={data.description} />
            <span className={styles.total}>
              <span className={styles.number}>{data.total || 0}</span> Clusters
            </span>
          </li>
        ))}
      </ul>
    );
  }
}
