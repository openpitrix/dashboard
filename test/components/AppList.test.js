import { BrowserRouter } from 'react-router-dom';

import AppList from 'components/AppList';

describe('AppList', () => {
  const props = {
    apps: [
      {
        app_id: 'app-lgmAg4qW13OQ',
        name: 'Wordpress',
        app_version_types: 'vmbased',
        company_name: 'QingClound',
        description: 'test',
        icon: ''
      }
    ],
    fixNav: true
  };

  it('basic render', () => {
    const wrapper = render(
      <BrowserRouter>
        <AppList {...props} />
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has search', () => {
    const newProps = {
      search: 'app',
      apps: []
    };
    const wrapper = render(
      <BrowserRouter>
        <AppList {...Object.assign(props, newProps)} />
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
