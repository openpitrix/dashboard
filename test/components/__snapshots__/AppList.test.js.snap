// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppList basic render 1`] = `
<div
  class="appList"
>
  <div
    class="title"
  >
    Newest
  </div>
  <div
    class="card card"
  >
    <div
      class="icon"
    >
      <img
        alt="Icon"
        class="img"
        data-origin-url=""
        src=""
      />
    </div>
    <div
      class="name"
    >
      Wordpress
    </div>
    <pre
      class="desc"
      title="test"
    >
      test
    </pre>
    <div
      class="attrs"
    >
      <dl>
        <dt
          class="label"
        >
          Service provider:
        </dt>
        <dd
          class="val"
        >
          QingClound
        </dd>
      </dl>
      <dl>
        <dt
          class="label"
        >
          Delivery type:
        </dt>
        <dd
          class="val"
        >
          <div
            class="versionType"
          >
            <label>
              VM
            </label>
          </div>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`AppList has search 1`] = `
<div
  class="appList"
>
  <div
    class="title"
  >
    search_results
  </div>
  <div
    class="noData"
  >
    NO_SEARCH_DATA
  </div>
</div>
`;
